openedx/features/offline_mode/tasks.py

```diff
@@ -23,11 +24,26 @@ def generate_offline_content_for_course(course_id):
     """
     course_key = CourseKey.from_string(course_id)
 
+    tasks = []
+
     for offline_supported_block_type in OFFLINE_SUPPORTED_XBLOCKS:
         for xblock in modulestore().get_items(course_key, qualifiers={'category': offline_supported_block_type}):
             is_not_closed = not hasattr(xblock, 'closed') or not xblock.closed()
             if is_not_closed and is_modified(xblock):
-                generate_offline_content_for_block.apply_async([str(xblock.location)])
+                task = generate_offline_content_for_block.s(str(xblock.location))
```

Comment:
What if any task fails? The `save_course_size` won't execute at all.
I can suggest to add an error handling inside a task.